homeassistant/components/otbr/config_flow.py

```diff
@@ -144,10 +144,6 @@ async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResu
             for current_entry in current_entries:
                 if current_entry.source != SOURCE_HASSIO:
                     continue
-                if current_entry.unique_id != discovery_info.uuid:
-                    self.hass.config_entries.async_update_entry(
-                        current_entry, unique_id=discovery_info.uuid
-                    )
                 current_url = yarl.URL(current_entry.data["url"])
                 if (
                     current_url.host != config["host"]
```

Comment:
To resolve @MartinHjelmare comment.

```suggestion
                    current_entry.unique_id != discovery_info.uuid
                    or current_url.host != config["host"]
```

Should we even check if host is different? That shouldn't matter as we key on unique ID.